### PR TITLE
Refactor page structure

### PR DESCRIPTION
**Pull request by tombruijn**
Set up new directory and page structure for better organization of
documentation. Use namespaces for separate implementations (Ruby,
Elixir, Front-end) and try to fit the existing pages as best as
possible.

I recommend you run it locally and click through it. Some page to page links will be broken, these will be fixed by redirets on S3.

As discussed in https://github.com/appsignal/rfcs/pull/3
## Note

This PR leaves a lot of gaps, but is an improvement over the current version. It is a deployable version.

The refactor is not done yet, but please review this PR in the current state.
## Changes
- Update navigation sidebar
- Move pages to new structure
- Split a couple pages where content couldn't fit in the new location anymore. Such as Operating System support.
- Remove some very old and inaccessible pages.
## TODO
- [x] Setup redirects for current URLs to new structure
## Screenshots

How (part of) the navigation will look:

image

### CODE REVIEW

**Comment by tombruijn** [comment]
@wesoudshoorn are you okay with the small visual changes I made to the sidebar navigation? Less spacing between sections so more fits in the sidebar

**Comment by wesoudshoorn** [comment]
@tombruijn I will check it out, maybe I've got some improvements of my own to add 👍

**Comment by tombruijn** [comment]
@wesoudshoorn alright, don't go too crazy. Trying to limit this PR to page structure changes. This is one of many PRs to come.

File: source/stylesheets/modules/_side_nav.css.sass [hunk 1]
```diff
@@ -1,29 +1,32 @@
 .mod-side_nav
-  padding: 50px 40px 40px
-
-  h2
-    @extend %h-small
-    margin-bottom: 5px
-    a
-      color: inherit
+  padding: 40px 40px 40px
```
Comment (wesoudshoorn) [inline source/stylesheets/modules/_side_nav.css.sass]: I think this can lose 2 x `40px`

File: source/stylesheets/modules/_side_nav.css.sass [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+  font-size: 14px
```
Comment (wesoudshoorn) [inline source/stylesheets/modules/_side_nav.css.sass]: Can we use a variable `$font-size-large` or placeholder for this `%t-small`

File: source/stylesheets/modules/_side_nav.css.sass [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
         padding-left: 15px
         margin: 5px 0px
-        border-left: 1px solid rgba($font-subheading,0.6)
+        border-left: 1px solid rgba($font-subheading, 0.6)
 
-  a
+  h2
+    @extend %h-small
+    margin-bottom: 5px
+    font-size: 16px
```
Comment (wesoudshoorn) [inline source/stylesheets/modules/_side_nav.css.sass]: Same here, if we're extending a header but also setting a font-size, the `%h-small` is probably not complete :)

**Comment by tombruijn** [comment]
@wesoudshoorn updated the CSS. Hope you approve 😁

**Comment by tombruijn** [comment]
Added redirect rules generator for AWS S3. Run `rake redirects`. The output should be copied here: 

s3 management console 2016-10-28 17-03-05 1

You can see the redirect in action here: http://tom-demo.s3-website-eu-west-1.amazonaws.com/bla.yml

`bla.yml` will redirect to `bla.html`

---

I had the option to use a gem for this, but it seems like a trivial thing to add a gem for. We don't deprecate pages that often, right? We could use the [configure-s3-website](https://github.com/laurilehmijoki/configure-s3-website#configuring-redirects) gem if the favor lies with automatic updating.

**Pull request merged**